### PR TITLE
Allow on_rightclick to be overidden

### PR DESCRIPTION
**Pull request by Ryan-Nolan**
Allow mods such as protection mods to over ride on_rightclick.
Usecase is creating shared doors without the need for keys

As far as im aware the only thing that needed completing was the API docs.

Reference: #2009

### CODE REVIEW

File: game_api.txt [hunk 1]
```diff
@@ -159,7 +159,7 @@ Doors API
 ---------
 
 The doors mod allows modders to register custom doors and trapdoors.
-
+
```
Comment (SmallJoker) [inline game_api.txt]: Accidental whitespace. Needs removal

File: game_api.txt [hunk 2]
```diff
@@ -218,6 +225,7 @@ The doors mod allows modders to register custom doors and trapdoors.
 	sound_open = sound play for open door, -- optional
 	sound_close = sound play for close door, -- optional
 	protected = false, -- If true, only placer can open the door (locked for others)
+	on_rightclick = function(pos, node, clicker, itemstack, pointed_thing) -- function containing the on_rightclick callback
```
Comment (SmallJoker) [inline game_api.txt]: Line is too long. Move comment to next line

File: mods/doors/init.lua [hunk 3]
```diff
@@ -2,9 +2,8 @@
 doors = {}
 
 -- private data
-local _doors = {}
-_doors.registered_doors = {}
-_doors.registered_trapdoors = {}
+doors.registered_doors = {}
+doors.registered_trapdoors = {}
```
Comment (SmallJoker) [inline mods/doors/init.lua]: These are now public. Please document them.

Comment (Ryan-Nolan) [reply mods/doors/init.lua]: Where should i list them, i cant find any other places where tables are listed/defined?

Comment (SmallJoker) [reply mods/doors/init.lua]: Here's an example of another table with registered items:
https://github.com/minetest/minetest_game/blob/1b10a98e69538b061bacc243759a951ad2ec0ddc/game_api.txt#L319-L320
https://github.com/minetest/minetest_game/blob/1b10a98e69538b061bacc243759a951ad2ec0ddc/game_api.txt#L337-L346
However, latter part is already done (`Door definition`, `Trapdoor definition`)

Comment (Ryan-Nolan) [reply mods/doors/init.lua]: thanks il get it updated now.

**Comment by paramat** [comment]
@sofar gave :+1: on IRC, see link above.

File: game_api.txt [hunk 4]
```diff
@@ -206,6 +219,8 @@ The doors mod allows modders to register custom doors and trapdoors.
 	sound_open = sound play for open door, -- optional
 	sound_close = sound play for close door, -- optional
 	protected = false, -- If true, only placer can open the door (locked for others)
+	on_rightclick = function(pos, node, clicker, itemstack, pointed_thing) 
+	-- function containing the on_rightclick callback
```
Comment (SmallJoker) [inline game_api.txt]: Needs a notice that it's optional - or - that it _defaults_ to a `doors.door_toggle`-wrapper function.

**Comment by SmallJoker** [comment]
One more thing to adjust. Will approve as soon it's done.

**Pull request merged**